### PR TITLE
Add CONTRIBUTING

### PR DESCRIPTION
**Pull request by vipulgupta2048**
- Add a sample Contributing document for new contributors to read and refer from. 
- Linked docs folder with contributing 
- Add CONTRIBUTING and License header in README
- Please suggest any changes. 
- Solves #164 

Signed-off-by: Vipul Gupta (@vipulgupta2048) <vipulgupta2048@gmail.com>

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# Contributing to Neural Network Intelligence (NNI)
```
[the remark this answers is not in the document]
Comment (QuanluZhang) [reply CONTRIBUTING.md]: hi @vipulgupta2048 , many thanks for your contribution :). this doc looks good, but I also suggest to move this file into `docs/` folder, we could add a link at the bottom of the `README.md` at the root level. Another suggestion is that could you add a link to `docs/HowToContribute.md` at the proper place in this file.

[the remark this answers is not in the document]
Comment (vipulgupta2048) [reply CONTRIBUTING.md]: Changes made as requested.

**Comment by vipulgupta2048** [comment]
Are there any style guidelines like PEP8, you guys follow. Or any naming conventions, some linters that travis has tests for. Any hard topics that you want to cover like making PR in separate branches, working with the team, how to you want to assign issues to contributors. Stuff like that. 

I write a lot of documentation. These are the usual things that goes into CONTRIBUTING.

**Comment by QuanluZhang** [comment]
@vipulgupta2048 , for your suggestion about lint. Yes, we have tslint and pylint, and are working on enabling them in ci :).

File: docs/CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+Provide PRs with appropriate tags for bug fixes or enhancements to the source code. Do follow the correct naming conventions and code styles when you work on and do try to implement all code reviews along the way.
+
+## Solve Existing Issues
+Head over to [issues](https://github.com/Microsoft/nni/issues) to find issues where help is needed from contributors. You can find issues tagged with 'good-first-issue' or 'help-wanted' to contribute in. 
+
+A person looking to contribute can take up an issue by claiming it as a comment/assign their Github ID to it. In case there is no PR or update in progress for a week on the said issue, then the issue reopens for anyone to take up again. We need to consider high priority issues/regressions where response time must be a day or so. 
+
+## Code Styles & Naming Conventions  
+< Need Help Here >
```
Comment (vipulgupta2048) [inline docs/CONTRIBUTING.md]: I need help here, don't merge it yet.

Comment (vipulgupta2048) [reply docs/CONTRIBUTING.md]: Should I just add pep8?

**Comment by yds05** [comment]
> Are there any style guidelines like PEP8, you guys follow. Or any naming conventions, some linters that travis has tests for. Any hard topics that you want to cover like making PR in separate branches, working with the team, how to you want to assign issues to contributors. Stuff like that.
> 
> I write a lot of documentation. These are the usual things that goes into CONTRIBUTING.

Hi, @vipulgupta2048 First, Thanks a lot for helping us draft the contributing document. Yes, we would like to cover the topics on how to send out PR (from your own forked repos to Microsoft/nni), merge PR(we prefer squash merge for cross-repository PR) ..etc

File: docs/CONTRIBUTING.md (hunk 2 above)
Comment (yds05) [reply docs/CONTRIBUTING.md]: @vipulgupta2048 Yes, we discussed and agreed that we could use pep8 for python code convention

**Comment by vipulgupta2048** [comment]
@yds05 @xuehui1991 @QuanluZhang Ready to go.

**Comment by vipulgupta2048** [comment]
Thank you, everyone, this has been my first major open-source contribution to Microsoft.
Quite small, though. Hoping to make it big. Thanks again :hatching_chick:

**Pull request merged**